chips/cc26xx/src/trng.rs

```diff
@@ -0,0 +1,158 @@
+//! TRNG - Random Number Generator for the cc26xx family
+//!
+//! Generates a random number using hardware entropy.
+//!
+
+use core::cell::Cell;
+use kernel::common::regs::{ReadOnly, ReadWrite, WriteOnly};
+use kernel::hil::rng;
+use prcm;
+
+#[repr(C)]
+struct RngRegisters {
+    out0: ReadOnly<u32>,
+    out1: ReadOnly<u32>,
+
+    irq_flag_stat: ReadOnly<u32, IrqStatus::Register>,
+    _irq_flag_mask: ReadOnly<u32>,
+    irq_flag_clr: WriteOnly<u32, IrqFlagClear::Register>,
+
+    ctl: ReadWrite<u32, Control::Register>,
+    cfg0: ReadWrite<u32, Config::Register>,
+
+    alarm_ctl: ReadWrite<u32, AlarmControl::Register>,
+
+    _fro_en: ReadOnly<u32>,
+    _fro_detune: ReadOnly<u32>,
+
+    _alarm_mask: ReadOnly<u32>,
+    _alarm_stop: ReadOnly<u32>,
+
+    _lfsr0: ReadOnly<u32>,
+    _lfsr1: ReadOnly<u32>,
+    _lfsr2: ReadOnly<u32>,
+
+    _r0: [u8; 0x1FB4],
+
+    sw_reset: ReadWrite<u32, SoftwareReset::Register>,
+}
+
+register_bitfields![
+    u32,
+    IrqStatus [
+        READY OFFSET(0) NUMBITS(1) []
+    ],
+    IrqFlagClear [
+        READY OFFSET(0) NUMBITS(1) []
+    ],
+    Control [
+        STARTUP_CYCLES  OFFSET(16) NUMBITS(16)  [],
+        TRNG_EN         OFFSET(10) NUMBITS(1)   []
+    ],
+    Config [
+        MAX_REFILL_CYCLES   OFFSET(16) NUMBITS(16) [],
+        SMPL_DIV            OFFSET(8)  NUMBITS(4)  [],
+        MIN_REFILL_CYCLES   OFFSET(0)  NUMBITS(8)  []
+    ],
+    AlarmControl [
+        // Alarm threshold for repeating pattern detectors
+        ALARM_THR   OFFSET(0)   NUMBITS(8) []
+    ],
+    SoftwareReset [
+        RESET OFFSET(0) NUMBITS(1) []
+    ]
+];
+
+const BASE_ADDRESS: *mut RngRegisters = 0x4002_8000 as *mut RngRegisters;
+
+pub static mut TRNG: Trng = Trng::new();
+
+pub struct Trng {
+    regs: *mut RngRegisters,
+    client: Cell<Option<&'static rng::Client>>,
+}
+
+impl Trng {
+    const fn new() -> Trng {
+        Trng {
+            regs: BASE_ADDRESS,
+            client: Cell::new(None),
+        }
+    }
+
+    pub fn enable(&self) {
+        // Ensure that the power domain TRNG resides in is enabled
+        if !prcm::Power::is_enabled(prcm::PowerDomain::Peripherals) {
+            prcm::Power::enable_domain(prcm::PowerDomain::Peripherals);
+
+            while !prcm::Power::is_enabled(prcm::PowerDomain::Peripherals) {}
+        }
+
+        // Setup the clock
```

Comment:
Add a comment or something to indicate that this implementation is blocking i.e, is not using interrupts!